src/model/guild/guild_id.rs

```diff
@@ -953,6 +953,27 @@ impl GuildId {
         http.as_ref().edit_guild_channel_positions(self.0, &Value::Array(items)).await
     }
 
+    /// Returns a list of [`Member`]s in a [`Guild`] whose username or nickname
+    /// starts with a provided string.
+    ///
+    /// Optionally pass in the `limit` to limit the number of results.
+    /// Minimum value is 1, maximum and default value is 1000.
+    ///
+    /// # Errors
+    ///
+    /// Returns an [`Error::Http`] if the API returns an error.
+    ///
+    /// [`Error::Http`]: crate::error::Error::Http
+    #[inline]
+    pub async fn search_members(
```

Comment:
Would it be possible to take a CacheHttp here and search the cache first? And in Guild's and PartialGuild's version of this method too

Reply:
The purpose of this method is to use the API feature, which will search by itself, especially for larger guilds. In my opinion it doesn't make sense to use the cache here

Reply:
Yeah I was thinking about that but another +1 to it might produce inconsistent/unexpected results.  Since if there are a few members cached that match, we still need to do an API request anyways for those additional non-cached that still can match.

It could work if it checked if all members are already cached (guild size == cached members size) to only use cache, but then at that point might as well use the other methods to be more explicit on intended behavior.

Reply:
I see. I thought the cache either contains either all or no members. If it can contain just a few users, that indeed seems too brittle to be worth it